src/pages/calendar/components/Section.tsx

```diff
@@ -59,9 +78,56 @@ export function SectionInfo({
   seat,
   meetingTimes,
 }: SectionInfoProps): JSX.Element {
-  const isASYNC = additionalNotes?.indexOf('asynchronous') !== -1;
-  const isSENG = additionalNotes?.indexOf('Reserved for BSENG students') !== -1;
-  const isCSC = additionalNotes?.indexOf('Reserved for students in a Computer Science program') !== -1;
+  const badges: BadgeProps[] = [];
+
+  // Special cases for async/sync/blended because UVic notes are funky
+  if (instructionalMethod === 'online') {
+    if (additionalNotes?.indexOf('mix of “real-time” and asynchronous') !== -1) {
+      badges.push({
+        name: 'Blended',
+        color: 'cyan',
+      });
+    } else if (additionalNotes?.indexOf('fully online and asynchronous') !== -1) {
+      badges.push({
+        name: 'Asynchronous',
+        color: 'cyan',
+      });
+    } else {
+      badges.push({
+        name: 'Synchronous',
+        color: 'cyan',
+      });
+    }
+  }
+
+  const rules: string[] = [
+    'Reserved for BSENG students',
+    'Computer Science program.',
+    'BEng students',
+    'Faculty of Engineering',
+    'SCIENCE students',
+    'BEng and BSEng students',
+    'BME, BSEN, CENG, ELEC students',
+  ];
+
+  const names: string[] = [
+    'SENG Only',
+    'CSC Only',
+    'ENGR Only',
+    'ENGR/CSC Only',
+    'SCI Only',
+    'BENG or BSENG only',
+    'BME/SENG/CENG/ELEC ONLY',
+  ];
+
+  for (let i = 0; i < rules.length; i++) {
+    if (additionalNotes?.indexOf(rules[i]) !== -1) {
+      badges.push({
+        name: names[i],
+        color: 'red',
+      });
+    }
+  }
```

Comment:
Could the `rules` and `names` be combined?
```js
const rules = [
  {rule: "Reserved for BSENG students", title: "SENG Only", color: "orange"}
]
const badges = rules.filter(r => additionalNotes?.indexOf(r) !== -1).map(r => {name: r.title, color: r.color})
```
Also, the rules/names should be defined outside the function. It doesn't make sense to declare them each time